plugin/rpcplugin/sandbox/sandbox_linux.go

```diff
@@ -0,0 +1,468 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sandbox
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"unsafe"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/mattermost/mattermost-server/plugin/rpcplugin"
+)
+
+func init() {
+	if len(os.Args) < 3 || os.Args[0] != "sandbox.runProcess" {
+		return
+	}
+
+	var config Configuration
+	if err := json.Unmarshal([]byte(os.Args[1]), &config); err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	if err := runProcess(&config, os.Args[2]); err != nil {
+		if eerr, ok := err.(*exec.ExitError); ok {
+			if status, ok := eerr.Sys().(syscall.WaitStatus); ok {
+				os.Exit(status.ExitStatus())
+			}
+		}
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func systemMountPoints() (points []*MountPoint) {
+	points = append(points, &MountPoint{
+		Source:      "proc",
+		Destination: "/proc",
+		Type:        "proc",
+	}, &MountPoint{
+		Source:      "/dev/null",
+		Destination: "/dev/null",
+	}, &MountPoint{
+		Source:      "/dev/zero",
+		Destination: "/dev/zero",
+	}, &MountPoint{
+		Source:      "/dev/full",
+		Destination: "/dev/full",
+	})
+
+	readOnly := []string{
+		"/dev/random",
+		"/dev/urandom",
+		"/etc/resolv.conf",
+		"/lib",
+		"/lib32",
+		"/lib64",
+		"/etc/ssl/certs",
+		"/system/etc/security/cacerts",
+		"/usr/local/share/certs",
+		"/etc/pki/tls/certs",
+		"/etc/openssl/certs",
+		"/etc/ssl/ca-bundle.pem",
+		"/etc/pki/tls/cacert.pem",
+		"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
+	}
+
+	for _, v := range []string{"SSL_CERT_FILE", "SSL_CERT_DIR"} {
+		if path := os.Getenv(v); path != "" {
+			readOnly = append(readOnly, path)
+		}
+	}
+
+	for _, point := range readOnly {
+		points = append(points, &MountPoint{
+			Source:      point,
+			Destination: point,
+			ReadOnly:    true,
+		})
+	}
+
+	return
+}
+
+func runProcess(config *Configuration, path string) error {
+	root, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(root)
+
+	if err := syscall.Mount("", "/", "", syscall.MS_PRIVATE|syscall.MS_REC, ""); err != nil {
+		return errors.Wrapf(err, "unable to make root private")
+	}
+
+	if err := mountMountPoints(root, systemMountPoints()); err != nil {
```

Comment:
Ok, correct me if I'm wrong here but the purpose of the system mount points is to give the process access to certain useful system directories inside the sandbox?

Reply:
Yeah. But some of these aren't just useful, but are actually strict requirements.

A lot of things break without /proc.

No Go program will run at all if it can't access /lib. For example, inspecting a test binary shows these dependencies:

```
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fd3b9d53000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fd3b9989000)
/lib64/ld-linux-x86-64.so.2 (0x00007fd3b9f70000)
```

Many Go programs will break if they don't have /dev/null (e.g. The exec package connects stdin/stdout to `os.DevNull` by default.).

You can't get good random numbers without /dev/urandom.

Nothing will be able to reach the Internet without /etc/resolv.conf.

You can't create temporary files without /tmp.